internal/server/edit_my_details.go

```diff
@@ -9,7 +9,7 @@ import (
 type EditMyDetailsClient interface {
 	MyDetails(sirius.Context) (sirius.MyDetails, error)
 	EditMyDetails(sirius.Context, int, string) error
-	HasPermission(sirius.Context, string, string) (bool, error)
+	GetMyPermissions(sirius.Context) (sirius.PermissionSet, error)
```

Comment:
The usual naming for getters is to not use `Get`, so `MyPermissions` here https://golang.org/doc/effective_go.html#Getters